src/features/filter.ts

```diff
@@ -1,12 +1,53 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Todo } from '../types/Todo';
 
 const initialState = {
   query: '',
   status: 'all',
 };
 
+type Status = 'all' | 'active' | 'completed';
+
+type State = {
+  query: string;
+  status: string;
```

Comment:
The `status` property in the `State` type should be typed as `Status` instead of `string` to ensure type safety and consistency with the `setStatus` reducer.